### ISSUE TITLE
`modal` close button and buttons not accessible on iOS

### ISSUE THREAD

**Issue by geospatialem** [Esri/calcite-components#4863]
### Summary

`modal` buttons, including the close button, are inaccessible on iOS devices while using VoiceOver, which traps the user in the dialog, unable to close or return to the page. All content, including the header is accessible, but all buttons are not navigable.

_Note:_ The `modal` works as-expected on desktop devices across screen readers and browsers allowing users to close, or execute button functionality as intended.

### Actual Behavior

On iOS devices the `modal` content is accessible, however the buttons, in particular the close button are inaccessible. The functionality is as-expected for desktop users, including screen reader users, however the same behavior is not experienced on mobile devices.

As a result, on iOS devices the user is trapped in the `modal`, and unable to exit, or return to the page content.

### Expected Behavior

All `modal` buttons should be accessible on iOS devices in addition to the content. Includes the ability to close and return to the page, or execute functionality associated with provided button(s), when applicable.

### Reproduction Sample

https://geospatialem.github.io/calcite-samples/modal.html

### Reproduction Steps

1. Open an iOS device (iPhone or iPad)
2. Open the Safari app
3. Open the [sample](https://geospatialem.github.io/calcite-samples/modal.html) in a new Safari tab
4. Enable VoiceOver in the device Settings (Settings > Accessibility > VoiceOver > On)
5. To access the modal, engage any of the modal buttons (e.g., the "Small scale modal" button)
6. Navigate the sample and try to close the `modal`
    - To navigate elements swipe left and/or right
    - To engage a button double tap the element when focused (similar to a tap for sighted users)
7. Notice the only element navigable is the header.
8. To disable VoiceOver, follow enable instructions changing VoiceOver to "Off"

### Reproduction Version

beta.83

### Working W3C Example/Tutorial

https://www.w3.org/WAI/ARIA/apg/example-index/dialog-modal/dialog

### Relevant Info

- iPhone 11 and 13
- iOS 15.5
- VoiceOver

### Actions needed
- [ ] Research solution(s)
- [ ] Implement solution(s)

**Comment by geospatialem** [comment]
This was solved with PR [4936](https://github.com/Esri/calcite-components/pull/4936/files#diff-1750b7bbce65a82bbddc6f7213b8f1cd41eec744c2d2ad386557f0d4d735e51b). 💯 

Verification confirmed on next.553 via [Codepen sample](https://codepen.io/geospatialem/pen/KKoQoGB).
